src/devices/src/legacy/i8042.rs

```diff
@@ -0,0 +1,38 @@
+use std::convert::TryInto;
+use vm_device::bus::{PioAddress, PioAddressOffset};
+use vm_device::MutDevicePio;
+use vm_superio::I8042Device;
+
+use utils::debug;
+
+use super::EventFdTrigger;
+
+//Offset of the command register, for write accesses (port 0x64).
+const COMMAND_OFFSET: u8 = 4;
+
+// Reset CPU command.
+const CMD_RESET_CPU: u8 = 0xFE;
+
+pub struct I8042Wrapper(pub I8042Device<EventFdTrigger>);
+
+impl MutDevicePio for I8042Wrapper {
+    fn pio_read(&mut self, _base: PioAddress, offset: PioAddressOffset, data: &mut [u8]) {
+        match offset.try_into() {
+            Ok(offset) => {
+                let read_data = self.0.read(offset);
+                data[0] = read_data;
+            }
+            Err(_) => debug!("Invalid I8042 read offset."),
+        }
+    }
+
+    fn pio_write(&mut self, _base: PioAddress, _offset: PioAddressOffset, data: &[u8]) {
+        if data.len() != 1 {
+            debug!("Invalid I8042 data length on write: {}", data.len());
+            return;
+        }
+        if let Err(_) = self.0.write(COMMAND_OFFSET, CMD_RESET_CPU) {
```

Comment:
By doing this, you always try to reset the cpu, even though the driver sent another command. Here we should just propagate the `offset` and `data[0]`.